misc/hawkbit-upload.py

```
@@ -828,17 +829,23 @@ def push_new_distribution_set_with_os(
             }
         ]
         
-        response = self.post("distributionsets", data)
-        self.id["newDistributionset"] = response[0]["id"] 
         
         if os_bundle_name is None:
             print("No bundle file provided to upload")
             return self.id["newDistributionset"], self.id["newSoftwaremodule"], None
         
         # Upload the artifact to the os software module
-        new_artifact_id = self.add_or_update_artifact(os_bundle_name, str(new_module_id))
-        
-        return self.id["newDistributionset"], self.id["newSoftwaremodule"], str(new_artifact_id)
+        try:
+            new_artifact_id = self.add_or_update_artifact(os_bundle_name, str(new_module_id))
+            response = self.post("distributionsets", data)
+            self.id["newDistributionset"] = response[0]["id"]
+            return self.id["newDistributionset"], self.id["newSoftwaremodule"], str(new_artifact_id)
+        except Exception as e:
+            #in case the upload of the arftifact fails, delete the software module
+            print(f"Error uploading the artifact {os.path.basename(os_bundle_name)} for module {software_module_name}, deleting module")
+            self.delete_softwaremodule(new_module_id)
+            return None, None, None
```

Are we failing the GitHub ci in this case? Or just silently swallowing the error?

---

@Mimoja Here we exit the python process with code 1, it should fail the CI

https://github.com/MeticulousHome/meticulous-machine/blob/c6165a08ce1268c8c3f636fc002b8d16a5b2501b/misc/hawkbit-upload.py#L956-L959